### PR TITLE
Equality

### PR DESCRIPTION
**Pull request by cthulahoops**
I'm assuming this isn't safe to merge yet.

### CODE REVIEW

File: vm/src/obj/objlist.rs [hunk 1]
```diff
@@ -120,8 +137,9 @@ fn reverse(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
 
 pub fn init(context: &PyContext) {
     let ref list_type = context.list_type;
+    list_type.set_attr("__eq__", context.new_rustfunc(list_eq));
     list_type.set_attr("__add__", context.new_rustfunc(list_add));
-    list_type.set_attr("__len__", context.new_rustfunc(len));
+    list_type.set_attr("__len__", context.new_rustfunc(list_len));
```
Comment (windelbouwman) [inline vm/src/obj/objlist.rs]: I think I made the same change, so no conflict :smile:

File: vm/src/obj/objtype.rs [hunk 2]
```diff
@@ -121,7 +120,7 @@ pub fn type_call(vm: &mut VirtualMachine, mut args: PyFuncArgs) -> PyResult {
     Ok(obj)
 }
 
-pub fn get_attribute(vm: &mut VirtualMachine, obj: PyObjectRef, name: &String) -> PyResult {
+pub fn get_attribute(vm: &mut VirtualMachine, obj: PyObjectRef, name: &str) -> PyResult {
```
Comment (windelbouwman) [inline vm/src/obj/objtype.rs]: Is it better to use str than String? I did not yet read the difference yet.

**Comment by windelbouwman** [comment]
Nice job! This is a good step towards more use of types.

File: vm/src/objbool.rs [hunk 3]
```diff
@@ -52,7 +68,7 @@ fn bool_str(vm: &mut VirtualMachine, args: PyFuncArgs) -> Result<PyObjectRef, Py
     let s = if v {
         "True".to_string()
     } else {
-        "True".to_string()
+        "False".to_string()
```
Comment (windelbouwman) [inline vm/src/objbool.rs]: Wow, tricky mistake well fixed!

**Comment by OddBloke** [comment]
What's invalid about the json types?

**Pull request merged**